compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorDistribute.cpp

```diff
@@ -33,347 +22,25 @@
 
 #define DEBUG_TYPE "iree-llvmgpu-vector-distribute"
 
-using LayoutDimension = mlir::iree_compiler::IREE::VectorExt::LayoutDimension;
-using LayoutDimensionAttr =
-    mlir::iree_compiler::IREE::VectorExt::LayoutDimensionAttr;
-using VectorLayoutInterface =
-    mlir::iree_compiler::IREE::VectorExt::VectorLayoutInterface;
-using PerDimLayoutAttr = mlir::iree_compiler::IREE::VectorExt::PerDimLayoutAttr;
-using LayoutAttr = mlir::iree_compiler::IREE::VectorExt::LayoutAttr;
-
 namespace mlir::iree_compiler {
 
 namespace {
 
-// Vector layout option setter aimed at contractions. Currently this only sets
-// anchors for two types of operations; vector.contract and vector.transfer_read
-// from non-shared memory. The assumption in this case is that all IR input to
-// this pass has a leaf rooted on a transfer_read or includes a contraction in
-// the program slice, meaning all operations should receive layouts. Layout
-// setting for other problems like reductions is TODO.
 class ContractionVectorLayoutOptions : public VectorLayoutOptions {
 public:
-  ContractionVectorLayoutOptions(Operation *root,
-                                 ArrayRef<int64_t> workgroupSize,
-                                 IREE::GPU::MMAScheduleAttr schedule,
-                                 Value laneId, int64_t subgroupSize,
-                                 bool printLayout)
-      : VectorLayoutOptions(root, /*fullConversion=*/!printLayout),
-        workgroupSize(workgroupSize), schedule(schedule),
-        printLayout(printLayout), patterns(root->getContext()) {
+  ContractionVectorLayoutOptions(Operation *root, Value laneId,
+                                 int64_t subgroupSize)
+      : VectorLayoutOptions(root), patterns(root->getContext()) {
```

Comment:
NIT: Seems like we don't need root as part of the vectorLayoutOptions or ContractionVectorLayoutOptions anymore? Maybe we can remove it, but we can do that in following PRs.

Reply:
Yeah we probably don't need VectorLayoutOptions anymore. I need to do a cleanup, can do after this patch.